src/__experimental__/components/date/date.component.js

```diff
@@ -133,19 +121,24 @@ class BaseDateInput extends React.Component {
 
   handleDateSelect = (selectedDate) => {
     const stringDateIso = DateHelper.formatDateString(selectedDate);
-
     this.isBlurBlocked = true;
     this.closeDatePicker();
-    this.updateVisibleValue(stringDateIso);
-    this.emitOnChangeCallback(stringDateIso);
+    this.updateVisibleValue(stringDateIso, true);
   };
 
-  updateVisibleValue = (date) => {
+  updateVisibleValue = (date, pickerUsed) => {
     const visibleValue = formatDateToCurrentLocale(date);
 
     this.setState({
       selectedDate: date,
       visibleValue
+    }, () => {
+      if (pickerUsed) {
+        const event = {
+          target: this.input
```

Comment:
It looks like we create these kinds of "synthesised event objects" quite often in various components: `event = { target: … }`

Maybe it'd be good to have a helper/utility function to DRY this up and to clarify its purpose? eg:
```js
/** @see https://developer.mozilla.org/en-US/docs/Web/API/Event/target */
function makeEventFromTarget(targetElement) {
  return { target: targetElement };
}

this.emitOnChangeCallback(makeEventFromTarget(this.input), date);
```

Reply:
Yep agreed I'm currently doing such an approach in the refactor now the format for emitted events has been agreed upon